File: C_Programming/Practise_Programs/PP_108.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_108
//
//  Description       : This program demonstrates string input using
//                      scanf() and highlights the issue with %s when
//                      reading a string containing spaces.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 13/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(n)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

int main()
{
    char Arr[50] = {'\0'};

    printf("Enter string : \n");
    scanf("%s", Arr);       // ISSUE

    printf("Enter string is : %s\n", Arr);
    
    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Jay Ganesh
//
//  Sample Output :
//
//      Enter string :
//      Enter string is : Jay
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_108.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_108
//
//  Description       : This program accepts a string from the user
//                      and attempts to display its first three
//                      characters using array-style indexing.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 26/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(n)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class PP_108
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        String Arr = null;

        System.out.println("Enter String : ");
        Arr = sobj.nextLine();

        // ERROR: String does not support array-style indexing
        System.out.println(Arr[0]);
        System.out.println(Arr[1]);
        System.out.println(Arr[2]);

        sobj.close();
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Ganesh
//
//  Sample Output :
//
//      Compilation Error
//
//////////////////////////////////////////////////////////////////